src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/CreateExceptionRecord.java

```diff
@@ -25,19 +28,50 @@
 
     private final ExceptionRecordMapper mapper;
     private final CcdApi ccdApi;
+    private final List<String> jurisdictionsWithDuplicatePrevention;
 
     public CreateExceptionRecord(
         ExceptionRecordMapper mapper,
-        CcdApi ccdApi
+        CcdApi ccdApi,
+        @Value("${jurisdictions-with-duplicate-er-prevention}") final List<String> jurisdictionsWithDuplicatePrevention
     ) {
         this.mapper = mapper;
         this.ccdApi = ccdApi;
+        this.jurisdictionsWithDuplicatePrevention = jurisdictionsWithDuplicatePrevention;
     }
 
     /**
-     * Creates an Exception Record from given envelope.
+     * Creates an exception record from given envelope, unless an exception record
+     * already exists for this envelope.
      */
-    public void createFrom(Envelope envelope) {
+    public void tryCreateFrom(Envelope envelope) {
```

Comment:
'try' prefix does not look good for me, it implies an exception to be thrown. Maybe something like 'createIfPossible' or 'createIfAllowed' would be a better name.

Reply:
I had the opposite intention - try prefix suggests that the method may not succeed to do its thing and at the same time not throw exception. This convention is well-known in .NET, so I thought I'd apply it here.

Reply:
ok

Reply:
On a second thought, this name doesn't fully match the convention, as in case of a technical failure an exception will be thrown. How about `createIfNotNotExisting`?

Reply:
The original method name was fine be me

Reply:
The original method was just creating an exception method. Now it does it conditionally, so I think it would be ideal to reflect that in its name.